constitution.md

```diff
@@ -0,0 +1,232 @@
+
+## Collaboration Policy
+CWG’s standard communication channels and development policy is outlined as follows, and will likely be adopted for other 
+associated CHIPS Alliance working groups.
+
+## Communication Channels
+High level leadership directives will be published via mailing list, and minutes of these discussions will be posted on the 
+Chisel website.
+
+All technical development discussions will be done either on Github with Github issues, pull requests, and comments, or via 
+a recurring open development meeting with video conferencing. Open development meetings will be run by project 
+administrators or senior reviewers.
+
+User-facing support is primarily through answering questions on **Stack Overflow**, contributing to and maintaining a **Github 
+Wiki page**, and promoting community dialogue with an associated **Gitter** channels. The purpose of Gitter is to move discussion 
+from mailing lists to a more modern communication outlet.
+
+
+### Development Policy
+Generally, development on mature products will release minor/major versions, where minor versions are backwards compatible, 
+and major versions can break ok API’s. Old API’s will be deprecated prior to removal.
+
+New features which are experimental could be released in an **experimental** package, whereby all API’s in this package are 
+subject to API changes without deprecation, and may change on a minor release (although this will be avoided if possible). 
+Successful experimental features/APIs will eventually be migrated to other packages.
+
+All code contributions to mature projects (Chisel, FIRRTL, etc.) must undergo a review process and pass continuous 
+integration tests prior to merging into the master branch.
+
+## Amending the Constitution
+Changes to the constitution are proposed as pull requests against this document.
+They are approved by 2/3 vote of the TAC as expressed by TAC members approving
+the pull request on Github.
+
+## TAC Voting
+For events that require a vote of the TAC (eg. changing the constitution,
+admitting new projects, admitting new members, selecting or changing TAC chair,
+expelling projects, expelling members), votes may occur via TAC members approving
+(or not approving) Github pull requests against this repository. Voting periods for
+a given change are set by the chair to a length of time not less than 2 weeks.
+Votes are tallied and if approved, pull requests are merged by the TAC Chair.
+
+Because many decisions may be non-controversial or involve personnel decisions better
+discussed in a meeting than via pull requests, votes **excluding changing the
+constitution** may occur in official TAC meetings. These votes will be tallied by the
+TAC chair, and if approved, reflected as pull requests by the chair against this
+respository. The chair must mention in the pull request when the vote took place, and
+another member of the TAC must approve the pull request for it to be merged.
+
+## Code of Conduct (Contributors)
```

Comment:
Possibility to add a clause indicating that the overall goal of this CoC is to advance the technical development of the Chisel ecosystem. CoCs have been controversial in some projects where it has derailed technical development

Reply:
After some discussion, let's just take an existing code of conduct, which has been vetted, verbatim.